config/initializers/saml_settings.rb

```diff
@@ -1,34 +1,24 @@
 module SAML
 
-  cert_file = ENV["CERTIFICATE_FILE"]
-  key_file = ENV["KEY_FILE"]
-
-  CERTFILES_EXIST = cert_file && File.exists?(cert_file) &&
-                    key_file && File.exists?(key_file)
-  NO_LOGIN_MODE = Rails.env.test? || !CERTFILES_EXIST
+  CONFIG = Rails.application.config_for(:saml).freeze
   SETTINGS = OneLogin::RubySaml::Settings.new
 
-  if NO_LOGIN_MODE
-    msg = "No SAML settings configured, starting up in NO LOGIN MODE."\
-          " ID.me integration will not function."
-    Rails.logger.warn msg
-  else
+  unless Rails.env.test?
```

Comment:
Again, here, I would try to put these into environment specific files so that the conditional logic is not necessary.

As of Rails 4.2 there is the `config_for` option, which makes the use of Figaro almost redundant or unnecessary. I'm willing to bet when Prescriptions used Figaro, they probably didn't know of the new feature: http://apidock.com/rails/v4.2.1/Rails/Application/config_for

It even supports having ERB in the YAML right out of the box.